### PR TITLE
refactor: upgrade various dependencies

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out Leather build 688e363 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9298925613), [Test report](https://leather-wallet.github.io/playwright-reports/refactor-package-upgrades-before-refactor), [Storybook](https://refactor-package-upgrades-before-refactor--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=refactor-package-upgrades-before-refactor)

Some general updates before I start with the `@stacks/*` packages

## Summary by CodeRabbit

- **Documentation**
  - Updated terminology in documentation from "browser extension" to "Web Extension."
  - Refined description of script context.
  - Removed instructions for running the application in development mode.

- **Dependencies**
  - Upgraded various package versions to ensure compatibility and security.

- **Bug Fixes**
  - Improved type imports for better maintainability and reduced bundle size.
  
- **Refactor**
  - Streamlined type declarations and imports across several components for consistency and clarity.
  - Renamed and reorganized types for better code readability and maintainability.

### CODE REVIEW

**Comment by kyranjamie** [comment]
Next I'll tackle upgrades to `@scure/*` and `@noble/*` that'd be helpful to complete before further refactoring, though this first requires monorepo changes.

**Review by alter-eggo — APPROVED** [review]

**Review by pete-watters — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -362,7 +361,13 @@
...
     "socket.io-parser": "4.2.4",
     "wrap-ansi": "7.0.0",
-    "webpack-dev-middleware": "5.3.4"
+    "webpack-dev-middleware": "5.3.4",
+    "eslint": "8.56.0"
+  },
+  "pnpm": {
+    "overrides": {
+      "eslint": "8.56.0"
+    }
```
Comment (edgarkhanzadian) [inline package.json:371]: Why pnpm overrides here?

**Review by edgarkhanzadian — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -260,11 +259,12 @@
...
-    "@pmmmwh/react-refresh-webpack-plugin": "0.5.11",
+    "@pmmmwh/react-refresh-webpack-plugin": "0.5.13",
     "@redux-devtools/cli": "4.0.0",
     "@redux-devtools/remote": "0.9.3",
     "@schemastore/web-manifest": "0.0.6",
-    "@sentry/webpack-plugin": "2.14.3",
+    "@sentry/react": "7.106.0",
+    "@sentry/webpack-plugin": "2.17.0",
     "@stacks/connect-react": "22.2.0",
     "@stacks/stacks-blockchain-api-types": "6.3.4",
```
Comment (fbwoolf) [inline package.json:270]: Should this one be updated too, I think it is also outdated?

**Review by fbwoolf — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (kyranjamie) [reply package.json:371]: It keeps installing version 9 without this

File: package.json (hunk 2 above)
Comment (kyranjamie) [reply package.json:270]: I will do separately, think there are many type errors

**Pull request merged**